### PR TITLE
Changed Filerator documentation to chpldoc format

### PR DESCRIPTION
**Pull request by bradcray**
Added chpldoc documentation to the four filerators: walkdirs, glob, findfiles, and listdir().

Successfully resisted the impulse to add formal types and return types to the code signatures due to being past code freeze (forgot to do this earlier).

### CODE REVIEW

**Comment by bradcray** [comment]
@lydia-duncan said she would review this

File: modules/standard/FileSystem.chpl [hunk 1]
```diff
@@ -1139,28 +1150,29 @@ proc locale.umask(mask: int): int {
...
+   :arg depth: indicates the maximum recursion depth to use (defaults to `max(int)`)
+   :type depth: int
 
+   :arg hidden: indicates whether to descend into hidden directories (defaults to `false`)
+   :type hidden: bool
+
+   :arg followlinks: indicates whether to follow symbolic links (defaults to `false`)
+   :type followlinks: bool
+
+   :arg sort: indicates whether or not to consider subdirectories in sorted order (defaults to `false`).  Note that requesting sorting has no effect in parallel invocations.
```
Comment (lydia-duncan) [inline modules/standard/FileSystem.chpl]: Note that you can wrap this line, either to line up with ':arg' or 'indicates', and the html generation will still be the same.

File: modules/standard/FileSystem.chpl [hunk 2]
```diff
@@ -509,14 +509,20 @@ proc exists(name: string): bool {
...
 
 
-/* iter findfiles(startdir = ".", recursive=false, hidden=false)
+/* Finds files from a given start directory and yields their names,
+   similar to simple invocations of the command-line `find` utility.
+   May be invoked in serial or non-zippered parallel contexts.
 
-   findfiles() is a simple find-like utility implemented using the
-   above routines
+   :arg startdir: the root directory from which to start the search, defaults to "."
```
Comment (lydia-duncan) [inline modules/standard/FileSystem.chpl]: The display for "." is a little weird.  I wonder if surrounding it with backticks will improve it?

Comment (lydia-duncan) [inline modules/standard/FileSystem.chpl]: For consistency, I've been capitalizing the first word in every description.

File: modules/standard/FileSystem.chpl [hunk 3]
```diff
@@ -1139,28 +1150,29 @@ proc locale.umask(mask: int): int {
...
-  
-   by default, walkdirs() will start in the current directory, process
-   directories in preorder; recursively traverse subdirectories; and
-   neither follow dotfile directories nor symbolic links.  It will not
-   sort the directories by default.
-*/
+/* Recursively walk a directory structure, yielding directory names.
+   May be invoked in serial or non-zippered parallel contexts.
+
+   :arg path: the directory from which to start the walk (defaults to ".")
```
Comment (lydia-duncan) [inline modules/standard/FileSystem.chpl]: The display comment I made above also applies to this use of "."

**Comment by lydia-duncan** [comment]
You can take or leave the comments as you see fit.  Thanks!

**Comment by bradcray** [comment]
@lydia-duncan : Incorporated changes based on your feedback.  The one other change I made this morning which I mentioned on the way in was to add the following note to walkdirs():

Note: The current parallel version is not very adaptive/dynamic in its application of parallelism to the list of subdirectories at any given level of the traversal, and could be improved in this regard.

Let me know if you're still OK with me committing.

File: modules/standard/FileSystem.chpl [hunk 4]
```diff
@@ -1139,28 +1155,33 @@ proc locale.umask(mask: int): int {
...
-   directories in preorder; recursively traverse subdirectories; and
-   neither follow dotfile directories nor symbolic links.  It will not
-   sort the directories by default.
-*/
+/* Recursively walk a directory structure, yielding directory names.
+   May be invoked in serial or non-zippered parallel contexts.  
+
+   Note: The current parallel version is not very adaptive/dynamic in
+   its application of parallelism to the list of subdirectories at any
+   given level of the traversal, and could be improved in this regard.
```
Comment (lydia-duncan) [inline modules/standard/FileSystem.chpl]: This could be an rst note if you wanted:

.. note::
   The current ...

Comment (lydia-duncan) [inline modules/standard/FileSystem.chpl]: There should be indentation in the last line.  I think you get my drift

**Comment by lydia-duncan** [comment]
:+1:

**Pull request merged**